scripts/vcf_site_level_filtering_wdl/JointVcfFiltering.wdl

```diff
@@ -1,292 +1,280 @@
 version 1.0
 
-# This is a workflow for filtering a joint callset VCF using INFO level annotations (so filtering is at the site level).
-# Note that the input VCFs here may be sharded by genomic position which may be helpful for large cohorts. The script
-# will output the same number of shards that are input.
-# This portion of the filtering pipeline will assign a SCORE INFO field annotation to each site, but does not yet apply
-# the filtering threshold to the final VCF.
+# Workflow for scoring and optionally filtering a VCF based on site-level annotations using the
```

Comment:
I was thinking we could make allele-specific mode automatic---we just need to detect whether any annotations have `Number=A`. So this would get rid of the corresponding parameter. Added a TODO to the meta issue for this.

As for hard filtering, the fact that we need to specify this via `extra_args` is alluded to in the parameter meta documentation. I agree that it would be nice to elevate certain parameters, but then this raises the question of why those parameters were singled out. I think it's better from a development perspective to treat everything as equitably as possible----i.e., elevate everything or elevate nothing.

Currently, we do "elevate" `resource_args`, since this is shared by the extract and score tools and we want to ensure the same values are passed. We also elevate any file arguments that might need to be localized (e.g., `python_ scripts`, which in turn suggests that we elevate the corresponding `model_backend` argument). But then everything else is relegated to `extra_args`. This at least has the benefits of giving a more generic interface (in that using `extra_args` provides maximal freedom to change arguments in the future, etc.).

As for whether we should replicate documentation for e.g. hard filtering here----in a perfect world, yes, but in a world where replicated documentation quickly falls out of sync, I'd rather refer users to the original documentation. Hopefully reading those docs and looking at example JSONs where hard filtering is turned on should make it clear how this works.

Reply:
I agree we shouldn't be replicating documentation here, but I disagree that if we elevate some inputs we must elevate them all. But perhaps for this use case of an example WDL nothing needs to be elevated. We can leave that for the WDLs that will run a specific use case.